Packs/Zscaler/Integrations/Zscaler/Zscaler.py

```diff
@@ -1016,13 +1017,220 @@ def set_user_command(args):
+        content = {
+            'ID': int(response.get('id', '')),
+            'Name': response.get('name', ''),
+            'Type': response.get('type', ''),
+            'Description': response.get('description', ''),
+            'Addresses': response.get('addresses', []),
+            'IpCategories': response.get('ipCategories', []),
+            'Countries': response.get('countries', [])
+        }
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': content,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown("IP Destination group", content),
+            'EntryContext': {'Zscaler.IPDestinationGroup': content},
+        }
+    return entry
+
+
+def edit_ip_destination_group(args: dict):
+    payload = dict()
+    ip_group_id = str(args.get('ip_group_id', '')).strip()
+    check_url = f"/ipDestinationGroups/{ip_group_id}"
+    response_data = http_request('GET', check_url).json()
+    if response_data.get('id', 0) == 0:
+        raise Exception(f'RESOURCE NOT FOUND with ip_group_id. {ip_group_id}')
+    for key in response_data.keys():
+        if key in ('name',
+                   'type',
+                   'countries',
+                   'ipCategories',
+                   'addresses',
+                   'description',
+                   'isNonEditable'):
+            payload[key] = response_data[key]
+    name = args.get('name', '')
+    countries = argToList(args.get('countries', ''))
+    ip_categories = argToList(args.get('ip_categories', ''))
+    addresses = argToList(args.get('addresses', ''))
+    description = args.get('description', '')
+    is_non_editable = args.get('is_non_editable', False)
+    if name != '':
+        payload['name'] = name
+    if len(countries) > 0:
+        payload['countries'] = countries
+    if len(ip_categories) > 0:
+        payload['ipCategories'] = ip_categories
+    if len(addresses) > 0:
+        payload['addresses'] = addresses
+    if description != '':
+        payload['description'] = description
+    payload['isNonEditable'] = is_non_editable
+    try:
+        cmd_url = f"/ipDestinationGroups/{ip_group_id}"
+        json_data = json.dumps(payload)
+        response = http_request('PUT', cmd_url, json_data, DEFAULT_HEADERS).json()
+    except json.decoder.JSONDecodeError as exp:
+        return_error(f'Failed to execute zscaler-edit-ip-destination-group command. Error: {str(exp)}')
+    content = {
+        'ID': int(response.get('id', '')),
+        'Name': response.get('name', ''),
+        'Type': response.get('type', ''),
+        'Description': response.get('description', ''),
+        'Addresses': response.get('addresses', []),
+        'IpCategories': response.get('ipCategories', []),
+        'Countries': response.get('countries', [])
+    }
```

Comment:
DRY 😉

Reply:
confused :(

Reply:
DRY = Don't Repeat Yourself

This was a hint to use the same `get_contents` function.